### PR TITLE
br: Add goleak test for br integration test

### LINKED ISSUE #52696: Add goroutine leak test for br integration tests

### PR DESCRIPTION
**Pull request by RidRisR**
### What problem does this PR solve?

Issue Number: close #52696

Problem Summary:

### What changed and how does it work?

1. Add goleak test for br integration tests
2. Add a 30 seconds expiration time for http idleconnections
4. reduce the number of http clients in `stream_status`
5. clean up some opened rpc clients
6. ignored some harmless goroutine leaks

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/cmd/br/main_test.go [hunk 1]
```diff
@@ -20,8 +20,39 @@ import (
...
+	if !skipLeakTest {
+		goleak.VerifyTestMain(m,
+			goleak.IgnoreCurrent(),
+			goleak.IgnoreTopFunction("github.com/pingcap/tidb/br/pkg/utils.StartExitSingleListener.func1"),
+			goleak.IgnoreTopFunction("github.com/pingcap/tidb/br/pkg/utils.StartDynamicPProfListener.func1"),
+			goleak.IgnoreTopFunction("gopkg.in/natefinch/lumberjack%2ev2.(*Logger).millRun"),
+			goleak.IgnoreTopFunction("go.etcd.io/etcd/client/v3.waitRetryBackoff"),
+			goleak.IgnoreTopFunction("net/http.(*persistConn).writeLoop"),
+			goleak.IgnoreTopFunction("internal/poll.runtime_pollWait"),
+			goleak.IgnoreTopFunction("google.golang.org/grpc.(*ClientConn).WaitForStateChange"),
```
Comment (YuJuncen) [inline br/cmd/br/main_test.go:48]: It seems this implies a open gRPC connection, is it safe to ignore this?

File: br/cmd/br/main_test.go [hunk 2]
```diff
@@ -20,8 +20,39 @@ import (
...
+	skipLeakTest := false
+	var newArgs []string
+	for _, arg := range os.Args {
+		if arg == "--skip-goleak" {
+			skipLeakTest = true
+		} else {
+			newArgs = append(newArgs, arg)
+		}
+	}
+	os.Args = newArgs
```
Comment (YuJuncen) [inline br/cmd/br/main_test.go:37]: I will prefer make this flag an environment variable here. Which will be more flexible.

**Review by YuJuncen — COMMENTED** [review]
Rest LGTM

File: br/cmd/br/main_test.go (hunk 2 above)
Comment (RidRisR) [reply br/cmd/br/main_test.go:37]: I see no difference. It is just for test, no need to worried about performance or user friendly.
Also, there may be more than one `RUN_BR` command in one test. It is not so convenient to set and unset environment variables again and again.

File: br/cmd/br/main_test.go (hunk 1 above)
Comment (RidRisR) [reply br/cmd/br/main_test.go:48]: This error shows in G07 and G08, and if you add a 1 min sleep before goleak, it doesn't show up. I believe there is a timeout, Therefore it should be harmless.

File: br/cmd/br/main_test.go (hunk 2 above)
Comment (YuJuncen) [reply br/cmd/br/main_test.go:37]: Well, in fact we can use `SOME_ENV=1 run_br` to set environment variable for one subprocess. Anyway this is only a test case, you may go whatever you like.

**Review by YuJuncen — APPROVED** [review]

**Review by Leavrth — APPROVED** [review]

**Comment by YuJuncen** [comment]
`br/pkg/stream/stream_status.go:435:30: method StatusController.Close already declared at br/pkg/stream/stream_status.go:347:30`

**Comment by YuJuncen** [comment]
(You may run `make bazel_prepare && bazel build //br/pkg/... --//build:with_nogo_flag=true` locally to run the static checker)

**Pull request merged**